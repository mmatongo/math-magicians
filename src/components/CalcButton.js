import React from 'react';
import PropTypes from 'prop-types';

class CalcButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { text } = this.props;
    return (
      <div className="calc-buttons">{text}</div>
    );
  }
}

CalcButton.propTypes = {
  text: PropTypes.string.isRequired,
};

export default CalcButton;
