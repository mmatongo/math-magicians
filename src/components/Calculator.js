import React from 'react';
import CalcButton from './CalcButton';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      calcObj: {},
    };
    this.handleCalculate = this.handleCalculate.bind(this);
  }

  handleCalculate(calcObj) {
    this.setState({ calcObj: calculate(calcObj) });
  }

  render() {
    const { calcObj } = this.state;

    let screen = '0';
    if (Object.keys(calcObj).length === 0 || (calcObj.total && calcObj.next && calcObj.operator === null)) {
      screen = '0';
    } else {
      screen = calcObj.next != null ? calcObj.next : calcObj.total;
    }

    return (
      <div className="calc-body">
        <div className="calc-screen">
          {screen}
        </div>
        <div className="keyboard">
          <div className="key-digits">
            <CalcButton text="AC" />
            <CalcButton text="+/-" />
            <CalcButton text="%" />
            <CalcButton text="7" />
            <CalcButton text="8" />
            <CalcButton text="9" />
            <CalcButton text="4" />
            <CalcButton text="5" />
            <CalcButton text="6" />
            <CalcButton text="1" />
            <CalcButton text="2" />
            <CalcButton text="3" />
            <CalcButton text="0" />
            <div className="key-dot">
              <CalcButton text="." />
            </div>
          </div>
          <div className="operators">
            <CalcButton text="รท" />
            <CalcButton text="x" />
            <CalcButton text="-" />
            <CalcButton text="+" />
            <CalcButton text="=" />
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
