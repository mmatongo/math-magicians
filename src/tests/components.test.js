import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../components/Calculator';
import Nav from '../components/Nav';
import CalcButton from '../components/CalcButton';
import CalculatorPage from '../pages/CalculatorPage';
import Home from '../pages/Home';
import Quote from '../pages/Quote';

describe('Test that all components are rendering just fine.', () => {
  it('Render the Calculator', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render the Nav', () => {
    const tree = renderer
      .create(<Router><Nav /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render the CalcButton.', () => {
    const text = 'test';
    const onClick = jest.fn(() => {});
    const calcObj = { string: 'test' };
    const tree = renderer
      .create(<CalcButton text={text} onClick={onClick} mathObj={calcObj} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render the CalculatorPage.', () => {
    const tree = renderer
      .create(<CalculatorPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render the Home page.', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote page renders correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
