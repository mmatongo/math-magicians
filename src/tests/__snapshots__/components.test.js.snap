// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that all components are rendering just fine. Quote page renders correctly 1`] = `
<div
  className="quote-container"
>
  <h3>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </h3>
</div>
`;

exports[`Test that all components are rendering just fine. Render the CalcButton. 1`] = `
<button
  className="calc-buttons"
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Test that all components are rendering just fine. Render the Calculator 1`] = `
<div
  className="calc-body"
>
  <div
    className="calc-screen"
  >
    0
  </div>
  <div
    className="keyboard"
  >
    <div
      className="key-digits"
    >
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <div
        className="key-dot"
      >
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
    </div>
    <div
      className="operators"
    >
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Test that all components are rendering just fine. Render the CalculatorPage. 1`] = `
<div
  className="calculator"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calc-body"
  >
    <div
      className="calc-screen"
    >
      0
    </div>
    <div
      className="keyboard"
    >
      <div
        className="key-digits"
      >
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <div
          className="key-dot"
        >
          <button
            className="calc-buttons"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
      </div>
      <div
        className="operators"
      >
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="calc-buttons"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test that all components are rendering just fine. Render the Home page. 1`] = `
<section>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</section>
`;

exports[`Test that all components are rendering just fine. Render the Nav 1`] = `
<div
  className="nav-component"
>
  <h1>
    Math Magicians
  </h1>
  <nav
    className="nav"
  >
    <ul
      className="nav-list"
    >
      <li
        className="nav-items"
      >
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <span
          className="divider"
        >
           | 
        </span>
      </li>
      <li
        className="nav-items"
      >
        <a
          aria-current={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <span
          className="divider"
        >
           | 
        </span>
      </li>
      <li
        className="nav-items"
      >
        <a
          aria-current={null}
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
        <span
          className="divider"
        >
           | 
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
